eng/cake/dotnet.cake

```diff
@@ -367,48 +372,33 @@ Task("dotnet-diff")
 
 // Tasks for Local Development
 
-Task("VS-DOGFOOD")
-    .Description("Provisions .NET 6 and launches an instance of Visual Studio using it.")
-    .IsDependentOn("dotnet")
-    .Does(() =>
-    {
-        StartVisualStudioForDotNet6(null);
-    });
-
-Task("VS-NET6")
-    .Description("Provisions .NET 6 and launches an instance of Visual Studio using it.")
-    .IsDependentOn("Clean")
+Task("VS")
+    .Description("Provisions .NET 6, and launches an instance of Visual Studio using it.")
     .IsDependentOn("dotnet")
     .IsDependentOn("dotnet-buildtasks")
     .Does(() =>
     {
+        // We need to pack maui before we can open a custom sln
+        if (Argument<string>("pack", "false").StartsWith("t", StringComparison.InvariantCultureIgnoreCase))
+            RunTarget("Default");
+
         StartVisualStudioForDotNet6();
     });
 
-Task("VS-WINUI")
-    .Description("Provisions .NET 6 and launches an instance of Visual Studio with WinUI projects.")
-        .IsDependentOn("VS-NET6");
-    //  .IsDependentOn("dotnet") WINUI currently can't launch application with local dotnet
-    //  .IsDependentOn("dotnet-buildtasks")
-
-Task("VS-ANDROID")
-    .Description("Provisions .NET 6 and launches an instance of Visual Studio with Android projects.")
+Task("VS-CLEAN")
+    .Description("Provisions .NET 6, delete bin/obj, and launches an instance of Visual Studio using it.")
     .IsDependentOn("Clean")
```

Comment:
One thing I wondered, should the cake `Clean` target be optional somehow? A lot of times I make a tiny change and want to build again, but I don't want it nuking bin/obj. Should there be like a `--incremental`/`--do-not-clean` argument (or some other name)?

Maybe the `Clean` target could check some input value and just return early.

Reply:
Yea I had split those up into two separate targets.

VS: procures dotnet, builds maui, opens maui.slnf
VS-CLEAN: The same as VS except it does a recursive delete of the bin/obj directories

I probably get a little too excited adding different targets for permutations :-) I removed `VS-CLEAN` and just added a `--clean` parameter. AFAICT cake doesn't support just doing `--switch` you always have to pass it something :-/